internal/pkg/node/constructors.go

```diff
@@ -14,23 +14,13 @@ import (
 	"gopkg.in/yaml.v3"
 )
 
-var (
-	ConfigFile string
-)
-
-func init() {
-	conf := warewulfconf.Get()
-	if ConfigFile == "" {
-		ConfigFile = path.Join(conf.Paths.Sysconfdir, "warewulf/nodes.conf")
-	}
-}
-
```

Comment:
Removing ConfigFile from node means that this path ends up getting copied several times throughout the code. I understand the original ordering concern; but I don't think duplicating this code in a bunch of other places is the answer.

Reply:
In principle you are right *but* the `nodes.conf` should only be read via `node.Get()` and written via `node.Persist()` so this code is just copied two times and not several times.

Reply:
The path to nodes.conf is duplicated at

- https://github.com/warewulf/warewulf/pull/1569/files#diff-18635393846b27ad0a0b42766e25f0ee688658e3806278b477fc312a26e30530R34
- https://github.com/warewulf/warewulf/pull/1569/files#diff-18635393846b27ad0a0b42766e25f0ee688658e3806278b477fc312a26e30530R35
- https://github.com/warewulf/warewulf/pull/1569/files#diff-cbf7980ed402d328dd25a662ff88774f8d6629f9b89d70e5514df7d5033c8249R36
- https://github.com/warewulf/warewulf/pull/1569/files#diff-2c728dc8f5f1c5947be37b4ccddf31e5655624904fa2aa9001abb3abe53c31feR22
- https://github.com/warewulf/warewulf/pull/1569/files#diff-2c728dc8f5f1c5947be37b4ccddf31e5655624904fa2aa9001abb3abe53c31feR23
- https://github.com/warewulf/warewulf/pull/1569/files#diff-e1aef9e67db5d9c8dac8ef8b0005e650dfecd42740f20fff784721f14b3fc08eR114
- https://github.com/warewulf/warewulf/pull/1569/files#diff-e17cee5fa398c4a024a96998083c6b139e45c35d6f892e4896fbb86828c442e5R56